data_hub_api/docmaps/v2/sql/docmaps_index.sql

```diff
@@ -231,30 +231,22 @@ t_result_with_preprint_url_and_has_evaluations AS (
 
 t_result_with_initial_preprint_version AS (
   SELECT
-    * EXCEPT(elife_doi),
-
-    CASE 
-      WHEN elife_doi IS NULL
-        THEN CONCAT('10.7554/eLife.',manuscript_id)
-      ELSE 
-        elife_doi
-    END AS elife_doi,
+    *,
 
     CAST(result.position_in_overall_stage AS STRING) AS elife_doi_version_str,
 
     CASE
       WHEN manual_preprint_version IS NOT NULL
         THEN manual_preprint_version
-      WHEN preprint_url LIKE '%10.1101/%v%'
-        THEN REGEXP_EXTRACT(preprint_url, r'10\.\d{3,}.*v([1-9])')
-      WHEN preprint_url LIKE '%researchsquare.com/article/rs-%v%' OR preprint_url LIKE '%arxiv.org/abs/%v%'
+      WHEN REGEXP_CONTAINS(preprint_doi_url, r'osf')
```

Comment:
Would be good to talk about those changes as I think I don't have enough context.
(should have added it to this PR, as I think it is the main/first one)

Reply:
We made the extraction more generic. Instead of checking for bioRxiv. Research Square and arXiv, we are looking for a more common pattern that could also apply to other preprint servers.
We moved osf up, because it might match the more generic version extraction (e.g. it might contain `v1`)